backend/app/Application/Handlers/Issues/VoteIssueHandler.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Application\Handlers\Issues;
+
+use App\Application\Commands\Issues\VoteIssueCommand;
+use App\Application\Interfaces\WebSocketService;
+use App\Application\Services\CurrentUserService;
+use App\Domain\Entities\Issue;
+use App\Domain\Enums\IssueStatuses;
+use App\Domain\Enums\UserIssueStatuses;
+use App\Domain\Repositories\IssueRepository;
+use DomainException;
+
+class VoteIssueHandler
+{
+    public function __construct(
+        private CurrentUserService $currentUserService,
+        private IssueRepository $issueRepository,
+        private WebSocketService $webSocketService
+    ) { }
+    
+    public function handle(VoteIssueCommand $command): Issue
+    {
+        $issue = $this->issueRepository->findByNumber($command->getNumber());
+        
+        $number = $command->getNumber();
+        if (!$issue) {    
+            throw new DomainException("Issue $number not found");
+        }
+        
+        $userName = $this->currentUserService->getUser()->getName();
```

Comment:
De nuevo la misma duda sobre si $this->currentUserService->getUser() es no nulo, por más que es una validación que estes haciendo a nivel de middleware, no está de más que la capa de aplicación (donde el principal actor es el handler) se defienda de esa situación inconsistente con una aserción.